lib/WebSocketServer.js

```diff
@@ -126,15 +126,15 @@ util.inherits(WebSocketServer, EventEmitter);
 
 WebSocketServer.prototype.close = function(callback) {
   // terminate all associated clients
-  var error = null;
-  try {
-    for (var i = 0, l = this.clients.length; i < l; ++i) {
-      this.clients[i].terminate();
+  let error = null;
+
+  if (this.options.clientTracking) {
+    try {
+      for (const client of this.clients) client.terminate();
```

Comment:
The only change i'd make here is maybe move the try/catch to the inside of the loop so if one client fails to terminate we still try to kill the rest of them.